src/dtls_cid.c

```diff
@@ -0,0 +1,361 @@
+/* dtls_cid.c
```

Comment:
Let's rename this to `dtls.c` and use it to put all DTLS specific functions. You will also need to add it to CMakeList.txt and the visual studio projects.